internal/utils/utils.go

```diff
@@ -0,0 +1,18 @@
+package utils
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func WriteJSON[T any](w http.ResponseWriter, status int, data T, headers http.Header) error {
+	w.Header().Set("Content-Type", "application/json")
+
+	for key, value := range headers {
+		w.Header()[key] = value
+	}
+
+	w.WriteHeader(status)
+
+	return json.NewEncoder(w).Encode(data)
```

Comment:
We will receive plain text message. Data should be in struct and struct should have appropriate tags. 

Consider creating some wrapper struct.

Reply:
When we call json.NewEncoder(w).Encode(data) the JSON is created and written to the
http.ResponseWriter in a single step, which means there’s no opportunity to set HTTP
response headers conditionally based on whether the Encode() method returns an error or
not. Please use json.Marshal instead